Sources/GRPCHTTP2TransportNIOPosix/GRPCHTTP2ServerTransportBootstrapper.swift

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCCore
+import GRPCHTTP2Core
+import NIOCore
+import NIOHTTP2
+
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+extension ChannelPipeline.SynchronousOperations {
+  public typealias HTTP2ConnectionChannel = NIOAsyncChannel<HTTP2Frame, HTTP2Frame>
+  public typealias HTTP2StreamMultiplexer = NIOHTTP2Handler.AsyncStreamMultiplexer<
+    (NIOAsyncChannel<RPCRequestPart, RPCResponsePart>, EventLoopFuture<MethodDescriptor>)
+  >
+
+  public func configureGRPCHTTP2ServerTransportPipeline(
+    channel: any Channel,
+    configuration: GRPCHTTP2TransportNIOPosixConfiguration
+  ) throws -> (HTTP2ConnectionChannel, HTTP2StreamMultiplexer) {
+    let serverConnectionHandler = ServerConnectionManagementHandler(
+      eventLoop: self.eventLoop,
+      maxIdleTime: configuration.maxIdleTime,
+      maxAge: configuration.maxConnectionAge,
+      maxGraceTime: configuration.maxGraceTime,
+      keepaliveTime: configuration.keepaliveTime,
+      keepaliveTimeout: configuration.keepaliveTimeout,
+      allowKeepaliveWithoutCalls: configuration.allowClientKeepalivePingsWithoutCalls,
+      minPingIntervalWithoutCalls: configuration.minPingIntervalWithoutCalls
+    )
+    let flushNotificationHandler = GRPCServerFlushNotificationHandler(
+      serverConnectionManagementHandler: serverConnectionHandler
+    )
+    try self.addHandler(flushNotificationHandler)
+
+    let streamMultiplexer = try self.configureAsyncHTTP2Pipeline(mode: .server) { streamChannel in
+      return streamChannel.eventLoop.makeCompletedFuture {
+        let streamHandler = GRPCServerStreamHandler(
+          scheme: configuration.tls != nil ? .https : .http,
```

Comment:
There's definitely more to setting up TLS than just this, but I'm not sure how to set up the pipeline to support it, so this is an open question

Reply:
Because we want this helper to be common to both NIOPosix and NIOTS we don't want it to set up TLS (as it's done in different ways between the two).